src/OpaqueVal.h

```diff
@@ -117,41 +138,62 @@ class OpaqueVal : public Val {
      * @return the broker representation, or an error if serialization
      * isn't supported or failed.
      */
-    std::optional<BrokerData> Serialize() const;
+    [[deprecated("Remove in v7.1: use SerializeData instead")]] broker::expected<broker::data> Serialize() const;
+
+    /**
+     * @copydoc Serialize
+     */
+    std::optional<BrokerData> SerializeData() const;
 
     /**
      * Reinstantiates a value from its serialized Broker representation.
      *
      * @param data Broker representation as returned by *Serialize()*.
      * @return unserialized instances with reference count at +1
      */
-    static OpaqueValPtr Unserialize(BrokerDataView data);
+    [[deprecated("Remove in v7.1: use UnserializeData instead")]] static OpaqueValPtr Unserialize(
+        const broker::data& data);
+
+    /**
+     * @copydoc Unserialize
+     */
+    static OpaqueValPtr UnserializeData(BrokerDataView data);
 
     /**
      * @copydoc Unserialize
      */
-    static OpaqueValPtr Unserialize(BrokerListView data);
+    static OpaqueValPtr UnserializeData(BrokerListView data);
 
 protected:
     friend class Val;
     friend class OpaqueMgr;
 
+    /**
+     * @deprecated Override DoSerializeData instead.
```

Comment:
Could you also add the "Remove in v7.1" to all the `@deprecated` ones? I think Tim is grepping for that when doing the removal passes.